PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -318,22 +318,161 @@ def miniAOD_customizeCommon(process):
     for idmod in photon_ids:
         setupAllVIDIdsInModule(process,idmod,setupVIDPhotonSelection,None,False,task)
 
-    #---------------------------------------------------------------------------
-    #Adding  Boosted Subjets taus
+    #-- Adding boosted taus
     from RecoTauTag.Configuration.boostedHPSPFTaus_cfi import addBoostedTaus
     addBoostedTaus(process)
-    #---------------------------------------------------------------------------
-    #Adding tau reco for 80X legacy reMiniAOD
-    #make a copy of makePatTauTask to avoid labels and substitution problems
-    _makePatTausTaskWithTauReReco = process.makePatTausTask.copy()
-    #add PFTau reco modules to cloned makePatTauTask
     process.load("RecoTauTag.Configuration.RecoPFTauTag_cff")
-    _makePatTausTaskWithTauReReco.add(process.PFTauTask)
-    #replace original task by extended one for the miniAOD_80XLegacy era
+    process.load("RecoTauTag.Configuration.HPSPFTaus_cff")
+    #-- Adding customization for 94X 2017 legacy reMniAOD
+    from Configuration.Eras.Modifier_run2_miniAOD_94XFall17_cff import run2_miniAOD_94XFall17
+    _makePatTausTaskWithRetrainedMVATauID = process.makePatTausTask.copy()
+    _makePatTausTaskWithRetrainedMVATauID.add(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask)
+    process.hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT = process.hpsPFTauDiscriminationByVLooseIsolationMVArun2v1DBoldDMwLT.clone()
+    process.hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT.mapping[0].cut = cms.string("RecoTauTag_tauIdMVAIsoDBoldDMwLT2017v1_WPEff95")
+    _makePatTausTaskWithRetrainedMVATauID.add(process.hpsPFTauDiscriminationByVVLooseIsolationMVArun2v1DBoldDMwLT)
+    ## modify raw mva values
+    run2_miniAOD_94XFall17.toModify(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTraw, mvaName = "RecoTauTag_tauIdMVAIsoDBoldDMwLT2017v1")
+    run2_miniAOD_94XFall17.toModify(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTraw, mvaOpt  = "DBoldDMwLTwGJ")
```

Comment:
all existing modules should be modified in the same file where they were defined.
If I'm not mistaken, it should be happening in RecoTauTag/Configuration/python/HPSPFTaus_cff.py
This will also allow for a more transparent application of the modifier for run2_miniAOD_80XLegacy and run2_miniAOD_94XFall17, e.g. with a syntax like
```python
for e in [ run2_miniAOD_80XLegacy, run2_miniAOD_94XFall17]:
e.toModify(hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTraw, 
                   mvaName = "RecoTauTag_tauIdMVAIsoDBoldDMwLT2017v1", 
                   mvaOpt  = "DBoldDMwLTwGJ")
```
Readability can also be improved by not repeating toModify with the module name for every single variable.

At a glance, it seems like exactly the same modifications are applied in both cases.
For this, it may even be more straightforward to define one modifier, e.g. ```run2_tauLT2017v1_mva```, which can be added into both run2_miniAOD_80XLegacy and run2_miniAOD_94XFall17.
I'd leave this for some later PR in the master branch.

Reply:
Dear Slava,
I'm afraid that moving modification of tau-Ids from `miniAOD_tools.py` to the file where they are defined (`RecoTauTag/Configuration/python/HPSPFTaus_cff.py`) will change the RECO/AOD event content when RECO sequences are run with the `run2_miniAOD_94XFall17` era flag. It is what we want to avoid, no?

Reply:
@slava77 I'm also not sure about the quations by @mbluj . Slava, once you clarify I'm fine to upload the change that I've already made locally. I think the change requested for tauProducer_cfi.py is safe in an case.

Reply:
OK, but there is not protection to use this era with cmsDriver for RECO. However, I agree that it will be misuse, and that it is generally a good practice to have a modifier where modified modules are defined. Finally, as far as I know there is not many users of taus from RECOAOD nowadays (apart validation).

Reply:
Technically, there is no incompatibility issue even for RECO.

The "no change" restriction extends to existing production eras.
The production era for e.g. 2017 is Run2_2017 (alone).